src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -1204,20 +1330,21 @@ class NodeConfig:
                     if cpuinfo['hyperthreads_per_core'] > 1:
                         pbs.logmsg(pbs.EVENT_DEBUG4,
                                    "Mapping hyperthreads to cores")
-                        cores = cpuinfo['cpu']
+                        cores = cpuinfo['cpu'].keys()
```

Comment:
xid and yid are used to construct a triagular matrix from which hyperthreads may be derived. The keys of cpuinfo['cpu'] are the actual core IDs. While we don't currently modify cpuinfo['cpu'] entries during this process, it was safer to go with independent lists that are known to be static.

Reply:
I'm not sure I understand.  What are you worried about changing?  If you say 'for key in <dict>', each value of key will be an index in the dictionary.  Are you worried that if we change key that it'll change something in the dictionary?  I don't think that is the case.

Reply:
The difference in cost in terms of memory and performance is negligible. Since we don't have any tests for this area of the code yet, I'm apprehensive toward changing it.